packages/wdio-sauce-service/package.json

```diff
@@ -23,13 +23,23 @@
     "url": "https://github.com/webdriverio/webdriverio/issues"
   },
   "dependencies": {
+    "@types/mocha": "^8.0.3",
     "@wdio/logger": "6.6.0",
     "@wdio/utils": "6.7.0",
-    "saucelabs": "^4.2.0"
+    "saucelabs": "^4.5.1"
   },
   "peerDependencies": {
     "@wdio/cli": "^6.0.1"
   },
+  "devDependencies": {
+    "webdriverio": "6.7.0",
+    "webdriver": "6.7.0",
+    "@wdio/cucumber-framework": "6.7.0",
+    "@wdio/jasmine-framework": "6.7.0",
+    "@wdio/reporter": "6.7.0",
+    "@types/cucumber": "6.0.1",
+    "cucumber": "6.0.5"
+  },
```

Comment:
This will be problematic if we release. As we are in a production environment it won't install dev dependencies and therefor the compile step will fail. I would suggest:

- remove `cucumber`
- move `@types/cucumber` into the dependencies
- reference all wdio types through tsconfig.json and use them through their namespace, e.g. WebDriver.DesiredCapabilities rather than importing them.

WDYT?

Reply:
Sounds good, I'll try it and see if it works and report back if I have issues.

Reply:
@christian-bromann by `reference all wdio types through tsconfig.json` you did mean to do it like that, right?:
```
"types": ["webdriverio", "webdriver", "@wdio/jasmine-framework" ,"@wdio/cucumber-framework", "@wdio/mocha-framework", "@wdio/utils"]
```
Or did you mean something else?

Because I'm getting an error, see `launcher.ts:21`:
```
TS2503: Cannot find namespace 'Webdriver'.
TS4063: Parameter 'capabilities' of constructor from exported class has or is using private name 'Webdriver'.
```

Reply:
ok, I guess this solution doesn't work then and we have to use references like [Jest](https://github.com/facebook/jest/blob/master/packages/jest-diff/tsconfig.json#L7-L12) does.

Reply:
I googled a bit and came to the same conclusion, I also found this:
https://medium.com/@NiGhTTraX/how-to-set-up-a-typescript-monorepo-with-lerna-c6acda7d4559
https://github.com/NiGhTTraX/ts-monorepo

In this case, I guess we need to change the root `tsconfig.json` to the following, right?:
```
{
    "extends": "./tsconfig.prod.json",
    "compilerOptions": {
        "inlineSourceMap": true,
        "baseUrl": ".",
        "paths": {
            "@wdio/*": ["packages/wdio-*/src"],
            "devtools": ["packages/devtools/src"],
            "eslint-plugin-wdio": ["packages/eslint-plugin-wdio/src"],
            "webdriver": ["packages/webdriver/src"],
            "werbdriverio": ["packages/webdriverio/src"]
        }
    }
}
```
If the `build` command runs fine, does this indicate it will work in production as well?

Reply:
Did you try specifying `types` in `package.json`? I hope this can solve typescript monorepo problem
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

Reply:
@hieuxlu the source code is right there for you to check yourself, but yes I did so ;)